### DISCUSSION TITLE
What is the goal of this collaboration?

### DISCUSSION THREAD

**Discussion by wbmlr** [karpathy/llm.c#358] [General]
How accurate is it to say, "The goal is to help everyone interested and take them through each and every step of llm development"?

A fully open source project because not all open source projects help simplify understanding or provide so much help.

I assume it's close to the explanations in the youtube videos on GPT.

How different is LLM development from GPT development? I am a noob. So, simple explanations will suffice.

**Comment by karpathy** [comment]
GPT is a series of LLMs from OpenAI (GPT-1, 2, 3, 3.5, 4, ...). Llama is a series of LLMs from Meta. Claude is a series of LLMs from Anthropic, etc etc.

Goal is to have a simple, direct implementation of LLM training, without the use of a large library/framework, as we're relatively directly programming the GPU hardware itself to make things as fast as possible. One effect of that is that it is fast, and might be practically useful. And yes another effect of that is that it can be quite educational, esp once I get around to building it from scratch again on video.

Hope that helps!

**Reply by youseai** [reply]
@karpathy I saw all the YouTube videos and still feel that I need a more basic understanding of things before I can deep dive into ongoing research. Can you recommend me some books where I can learn about it?

**Reply by dagelf** [reply]
@youseai Presumably you have also gotten your hands dirty and played with NN's of different sizes and architectures to see what they are capable of and to get a feeling for how they work? From my experience it's really hard and rare to find a book or even a course that can provide the same depth of understanding as just trying it and changing things. (I'm even confident that many if not most of the LLM experts, don't *really* grok why/how it works :sweat_smile: - please convince me otherwise!)
